drivers/ethernet/eth_native_posix_adapt.c

```diff
@@ -138,3 +143,21 @@ ssize_t eth_write_data(int fd, void *buf, size_t buf_len)
 {
 	return write(fd, buf, buf_len);
 }
+
+#if defined(CONFIG_NET_GPTP)
+int eth_clock_gettime(struct net_ptp_time *time)
+{
+	struct timespec tp;
+	int ret;
+
+	ret = clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
```

Comment:
looking at this, I would actually it would be better to not access the host time directly, but to have the timer driver provide you the host time, to have just 1 central place for it.